apps/api/src/server/applications/search/search.validators.js

```diff
@@ -0,0 +1,66 @@
+// constants
+const validRoles = new Set (['inspector', 'case-officer', 'case-admin-officer']);
+const maxResultsPerPage = 20;
+const maxPages = 30;
+
+/** @type {import('express').RequestHandler } */
+export const validateSearchCriteria = async (request, response, next) => {
+
+	if (!validQuery(request.body.query)) {
+		response.status(400).send({
+			errors: {
+				status: 'Query cannot be blank'
+			}
+		});
+	} else if (!validRole(request.body.role)) {
+		response.status(403).send({
+			errors: {
+				status: 'Role is not valid'
+			}
+		});
+	} else if (!validPageNumber(request.body.pageNumber)) {
+		response.status(400).send({
+			errors: {
+				status: 'Page Number not in valid range'
+			}
+		});
+	} else if (!validPageSize(request.body.pageSize)) {
+		response.status(400).send({
+			errors: {
+				status: 'Page Size not in valid range'
+			}
+		});
+	} else {
+		next();
+	}
```

Comment:
Firstly, I would say that the role should be the first thing that is validated. Second, why are we not using `express-validator` here?